src/engines/kotor/gui/chargen/chargeninfo.cpp

```diff
@@ -201,7 +201,10 @@ void CharacterGenerationInfo::recreateHead() {
 	Graphics::Aurora::Model *head = _head;
 	_head = loadModelObject(Creature::getHeadMeshString(getGender(), getSkin(), getFace()), "");
 	_body->attachModel("headhook", _head);
+
+	GfxMan.lockFrame();
 	delete head;
+	GfxMan.unlockFrame();
```

Comment:
The attachModel() call should probably be moved within the locking as well

Reply:
We should probably have an RAII class for handling the lock/unlock (if we don't already).